cypress/integration/modal_spec.js

```diff
@@ -0,0 +1,34 @@
+describe('Modal Spec', () => {
+  beforeEach(() => {
+    cy.visit('http://localhost:3000/').get('.add-post-btn').click();
+  });
+
+  it('Should open a modal upon clicking Add a Post!', () => {
+    cy.get('.modal-content').should('be.visible');
+  });
+
+  it('Should contain an input form for title', () => {
+    cy.get('.post-title').should('be.visible');
+  });
+
+  it('Should contain an input form for description', () => {
+    cy.get('.post-description').should('be.visible');
+  });
+
+  it('Should contain an input to add images', () => {
+    cy.get('.upload__image-wrapper').should('be.visible');
+  });
+
+  it('Should contain an option to remove images', () => {
+    cy.get('.remove-img-btn').should('be.visible');
+  });
+
+  it('Should contain an submit button', () => {
+    cy.get('.post-submit-btn').should('be.visible');
+  });
+
+  it('Submit button should close modal and return to homescreen', () => {
+    cy.get('.post-submit-btn').click();
+    cy.get('.cards-container').should('be.visible');
+  });
```

Comment:
Do we give the user some sort of success message or is that just that they see the new post? Should we be testing that the newly created post is now present?

Reply:
Yes! That will come when I am actually stubbing some data in, which I am not yet... 
And i figured the closing of the modal and the appearance of the post was sufficient - at least that's something I've seen in the real world. Open to changing that!

Reply:
I think the closing of the modal and seeing the new post is potentially enough. There is a brief loading-type message in the modal while the mutation is processing. And yeah we will definitely test to see a new post created